Sources/TuistGenerator/Graph/TargetNode.swift

```diff
@@ -137,6 +136,42 @@ class TargetNode: GraphNode {
             return CocoaPodsNode.read(path: path.appending(podsPath), cache: cache)
         case let .package(packageType):
             return PackageNode(packageType: packageType, path: path)
+        case let .carthage(project, target):
+
+            let circularFrom = GraphCircularDetectorNode(path: path, name: name)
+            let projectPath = path.appending(project)
+            let circularTo = GraphCircularDetectorNode(path: projectPath, name: target)
+            circularDetector.start(from: circularFrom, to: circularTo)
+            
+            let targetNode = try TargetNode.read(
+                name: target,
+                path: projectPath,
+                cache: cache,
+                circularDetector: circularDetector,
+                modelLoader: modelLoader
+            )
+            
+            let platformName: String
+            
+            switch targetNode.target.platform {
+            case .iOS:
+                platformName = "iOS"
+            case .macOS:
+                platformName = "Mac"
+            case .tvOS:
+                platformName = "tvOS"
+            }
+            
+            let carthageBuild = path
+                .appending(project)
+                .appending(RelativePath("../../Build/\(platformName)/\(targetNode.target.productNameWithExtension)"))
```

Comment:
this will likely need some auto discovery since the project path might not be two levels deep.

E.g.

`Carthage/Checkouts/MyProject/Some/Deep/Folder`
`Carthage/Build`

We could possibly traverse to find the Cartfile and the Carthage directory and go from there ...